osu.Game/Beatmaps/Drawables/DifficultyIconTooltip.cs

```diff
@@ -70,8 +160,43 @@ public void SetContent(DifficultyIconTooltipContent content)
 
             displayedContent = content;
 
+            // Header row
             starRating.Current.BindTarget = displayedContent.Difficulty;
             difficultyName.Text = displayedContent.BeatmapInfo.DifficultyName;
+
+            double rate = 1;
+
+            if (displayedContent.Mods != null)
+            {
+                foreach (var mod in displayedContent.Mods.OfType<IApplicableToRate>())
+                    rate = mod.ApplyToRate(0, rate);
+            }
+
+            double bpmAdjusted = displayedContent.BeatmapInfo.BPM * rate;
+
+            BeatmapDifficulty originalDifficulty = new BeatmapDifficulty(displayedContent.BeatmapInfo.Difficulty);
+
+            if (displayedContent.Mods != null)
+            {
+                foreach (var mod in displayedContent.Mods.OfType<IApplicableToDifficulty>())
+                {
+                    mod.ApplyToDifficulty(originalDifficulty);
+                }
+            }
+
+            Ruleset ruleset = displayedContent.Ruleset.CreateInstance();
+            BeatmapDifficulty adjustedDifficulty = ruleset.GetRateAdjustedDisplayDifficulty(originalDifficulty, rate);
+
+            // Difficulty row
+            circleSize.Text = "CS: " + adjustedDifficulty.CircleSize.ToString("0.##");
+            drainRate.Text = " HP: " + adjustedDifficulty.DrainRate.ToString("0.##");
+            approachRate.Text = " AR: " + adjustedDifficulty.ApproachRate.ToString("0.##");
+            overallDifficulty.Text = " OD: " + adjustedDifficulty.OverallDifficulty.ToString("0.##");
+
+            // Misc row
+            length.Text = "Length: " + TimeSpan.FromMilliseconds(displayedContent.BeatmapInfo.Length / rate).ToString("mm\\:ss");
+            bpm.Text = " BPM: " + bpmAdjusted;
```

Comment:
No rounding or significant digit spec, will look broken

Reply:
Beatmap length is effectively rounded via the formatting (the toString). This constrains it to be MM:SS.
As for the bpm, do you think it would be better to round to the 2nd decimal place, or just to whole numbers?

Reply:
Probably round to whole numbers to match song select.